File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/task_run_summary.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class TaskRunSummary(object):
    """
    Summary information about a TaskRun.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new TaskRunSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this TaskRunSummary.
        :type key: str

        :param name:
            The value to assign to the name property of this TaskRunSummary.
        :type name: str

        :param task_key:
            The value to assign to the task_key property of this TaskRunSummary.
        :type task_key: str

        :param job_run_key:
            The value to assign to the job_run_key property of this TaskRunSummary.
        :type job_run_key: str

        :param parent_job_run_key:
            The value to assign to the parent_job_run_key property of this TaskRunSummary.
        :type parent_job_run_key: str

        :param root_job_run_key:
            The value to assign to the root_job_run_key property of this TaskRunSummary.
        :type root_job_run_key: str

        :param cluster_validation_start_time:
            The value to assign to the cluster_validation_start_time property of this TaskRunSummary.
        :type cluster_validation_start_time: int

        :param start_time:
            The value to assign to the start_time property of this TaskRunSummary.
        :type start_time: int

        :param end_time:
            The value to assign to the end_time property of this TaskRunSummary.
        :type end_time: int

        :param state:
            The value to assign to the state property of this TaskRunSummary.
        :type state: oci.aidataplatform_dp.models.State

        :param external_id:
            The value to assign to the external_id property of this TaskRunSummary.
        :type external_id: str

        :param retry_attempt:
            The value to assign to the retry_attempt property of this TaskRunSummary.
        :type retry_attempt: int

        """
        self.swagger_types = {
            'key': 'str',
            'name': 'str',
            'task_key': 'str',
            'job_run_key': 'str',
            'parent_job_run_key': 'str',
            'root_job_run_key': 'str',
            'cluster_validation_start_time': 'int',
            'start_time': 'int',
            'end_time': 'int',
            'state': 'State',
            'external_id': 'str',
            'retry_attempt': 'int'
        }

        self.attribute_map = {
            'key': 'key',
            'name': 'name',
            'task_key': 'taskKey',
            'job_run_key': 'jobRunKey',
            'parent_job_run_key': 'parentJobRunKey',
            'root_job_run_key': 'rootJobRunKey',
            'cluster_validation_start_time': 'clusterValidationStartTime',
            'start_time': 'startTime',
            'end_time': 'endTime',
            'state': 'state',
            'external_id': 'externalId',
            'retry_attempt': 'retryAttempt'
        }

        self._key = None
        self._name = None
        self._task_key = None
        self._job_run_key = None
        self._parent_job_run_key = None
        self._root_job_run_key = None
        self._cluster_validation_start_time = None
        self._start_time = None
        self._end_time = None
        self._state = None
        self._external_id = None
        self._retry_attempt = None

    @property
    def key(self):
        """
        Gets the key of this TaskRunSummary.
        The OCID of the TaskRun.


        :return: The key of this TaskRunSummary.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this TaskRunSummary.
        The OCID of the TaskRun.


        :param key: The key of this TaskRunSummary.
        :type: str
        """
        self._key = key

    @property
    def name(self):
        """
        Gets the name of this TaskRunSummary.
        A user-friendly name. Does not have to be unique, and is changeable.


        :return: The name of this TaskRunSummary.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this TaskRunSummary.
        A user-friendly name. Does not have to be unique, and is changeable.


        :param name: The name of this TaskRunSummary.
        :type: str
        """
        self._name = name

    @property
    def task_key(self):
        """
        Gets the task_key of this TaskRunSummary.
        The display name of the task. User can specify a value for this.


        :return: The task_key of this TaskRunSummary.
        :rtype: str
        """
        return self._task_key

    @task_key.setter
    def task_key(self, task_key):
        """
        Sets the task_key of this TaskRunSummary.
        The display name of the task. User can specify a value for this.


        :param task_key: The task_key of this TaskRunSummary.
        :type: str
        """
        self._task_key = task_key

    @property
    def job_run_key(self):
        """
        Gets the job_run_key of this TaskRunSummary.
        The OCID of the job.


        :return: The job_run_key of this TaskRunSummary.
        :rtype: str
        """
        return self._job_run_key

    @job_run_key.setter
    def job_run_key(self, job_run_key):
        """
        Sets the job_run_key of this TaskRunSummary.
        The OCID of the job.


        :param job_run_key: The job_run_key of this TaskRunSummary.
        :type: str
        """
        self._job_run_key = job_run_key

    @property
    def parent_job_run_key(self):
        """
        Gets the parent_job_run_key of this TaskRunSummary.
        The OCID of the job.


        :return: The parent_job_run_key of this TaskRunSummary.
        :rtype: str
        """
        return self._parent_job_run_key

    @parent_job_run_key.setter
    def parent_job_run_key(self, parent_job_run_key):
        """
        Sets the parent_job_run_key of this TaskRunSummary.
        The OCID of the job.


        :param parent_job_run_key: The parent_job_run_key of this TaskRunSummary.
        :type: str
        """
        self._parent_job_run_key = parent_job_run_key

    @property
    def root_job_run_key(self):
        """
        Gets the root_job_run_key of this TaskRunSummary.
        The OCID of the job.


        :return: The root_job_run_key of this TaskRunSummary.
        :rtype: str
        """
        return self._root_job_run_key

    @root_job_run_key.setter
    def root_job_run_key(self, root_job_run_key):
        """
        Sets the root_job_run_key of this TaskRunSummary.
        The OCID of the job.


        :param root_job_run_key: The root_job_run_key of this TaskRunSummary.
        :type: str
        """
        self._root_job_run_key = root_job_run_key

    @property
    def cluster_validation_start_time(self):
        """
        Gets the cluster_validation_start_time of this TaskRunSummary.
        The time at which the cluster validation started in epoch milliseconds.


        :return: The cluster_validation_start_time of this TaskRunSummary.
        :rtype: int
        """
        return self._cluster_validation_start_time

    @cluster_validation_start_time.setter
    def cluster_validation_start_time(self, cluster_validation_start_time):
        """
        Sets the cluster_validation_start_time of this TaskRunSummary.
        The time at which the cluster validation started in epoch milliseconds.


        :param cluster_validation_start_time: The cluster_validation_start_time of this TaskRunSummary.
        :type: int
        """
        self._cluster_validation_start_time = cluster_validation_start_time

    @property
    def start_time(self):
        """
        Gets the start_time of this TaskRunSummary.
        The time at which the job execution started in epoch milliseconds.


        :return: The start_time of this TaskRunSummary.
        :rtype: int
        """
        return self._start_time

    @start_time.setter
    def start_time(self, start_time):
        """
        Sets the start_time of this TaskRunSummary.
        The time at which the job execution started in epoch milliseconds.


        :param start_time: The start_time of this TaskRunSummary.
        :type: int
        """
        self._start_time = start_time

    @property
    def end_time(self):
        """
        Gets the end_time of this TaskRunSummary.
        The time at which the job execution started in epoch milliseconds.


        :return: The end_time of this TaskRunSummary.
        :rtype: int
        """
        return self._end_time

    @end_time.setter
    def end_time(self, end_time):
        """
        Sets the end_time of this TaskRunSummary.
        The time at which the job execution started in epoch milliseconds.


        :param end_time: The end_time of this TaskRunSummary.
        :type: int
        """
        self._end_time = end_time

    @property
    def state(self):
        """
        Gets the state of this TaskRunSummary.

        :return: The state of this TaskRunSummary.
        :rtype: oci.aidataplatform_dp.models.State
        """
        return self._state

    @state.setter
    def state(self, state):
        """
        Sets the state of this TaskRunSummary.

        :param state: The state of this TaskRunSummary.
        :type: oci.aidataplatform_dp.models.State
        """
        self._state = state

    @property
    def external_id(self):
        """
        Gets the external_id of this TaskRunSummary.
        The external ID of the task execution.


        :return: The external_id of this TaskRunSummary.
        :rtype: str
        """
        return self._external_id

    @external_id.setter
    def external_id(self, external_id):
        """
        Sets the external_id of this TaskRunSummary.
        The external ID of the task execution.


        :param external_id: The external_id of this TaskRunSummary.
        :type: str
        """
        self._external_id = external_id

    @property
    def retry_attempt(self):
        """
        Gets the retry_attempt of this TaskRunSummary.
        Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.


        :return: The retry_attempt of this TaskRunSummary.
        :rtype: int
        """
        return self._retry_attempt

    @retry_attempt.setter
    def retry_attempt(self, retry_attempt):
        """
        Sets the retry_attempt of this TaskRunSummary.
        Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.


        :param retry_attempt: The retry_attempt of this TaskRunSummary.
        :type: int
        """
        self._retry_attempt = retry_attempt

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/TaskRunSummary.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * Summary information about a TaskRun.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=TaskRunSummary.Builder.class)

public final class TaskRunSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "name", "taskKey", "jobRunKey", "parentJobRunKey", "rootJobRunKey", "clusterValidationStartTime", "startTime", "endTime", "state", "externalId", "retryAttempt"})
    public TaskRunSummary(String key, String name, String taskKey, String jobRunKey, String parentJobRunKey, String rootJobRunKey, Long clusterValidationStartTime, Long startTime, Long endTime, State state, String externalId, Integer retryAttempt) {
        super();
        this.key = key;
        this.name = name;
        this.taskKey = taskKey;
        this.jobRunKey = jobRunKey;
        this.parentJobRunKey = parentJobRunKey;
        this.rootJobRunKey = rootJobRunKey;
        this.clusterValidationStartTime = clusterValidationStartTime;
        this.startTime = startTime;
        this.endTime = endTime;
        this.state = state;
        this.externalId = externalId;
        this.retryAttempt = retryAttempt;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * The OCID of the TaskRun.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * The OCID of the TaskRun.
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("name")
private String name;

        /**
         * A user-friendly name. Does not have to be unique, and is changeable.
         * @param name the value to set
         * @return this builder
         **/
        

public Builder name(String name) {
    this.name = name;
    return this;
}
            /**
     * The display name of the task. User can specify a value for this.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("taskKey")
private String taskKey;

        /**
         * The display name of the task. User can specify a value for this.
         * @param taskKey the value to set
         * @return this builder
         **/
        

public Builder taskKey(String taskKey) {
    this.taskKey = taskKey;
    return this;
}
            /**
     * The OCID of the job.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("jobRunKey")
private String jobRunKey;

        /**
         * The OCID of the job.
         * @param jobRunKey the value to set
         * @return this builder
         **/
        

public Builder jobRunKey(String jobRunKey) {
    this.jobRunKey = jobRunKey;
    return this;
}
            /**
     * The OCID of the job.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("parentJobRunKey")
private String parentJobRunKey;

        /**
         * The OCID of the job.
         * @param parentJobRunKey the value to set
         * @return this builder
         **/
        

public Builder parentJobRunKey(String parentJobRunKey) {
    this.parentJobRunKey = parentJobRunKey;
    return this;
}
            /**
     * The OCID of the job.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("rootJobRunKey")
private String rootJobRunKey;

        /**
         * The OCID of the job.
         * @param rootJobRunKey the value to set
         * @return this builder
         **/
        

public Builder rootJobRunKey(String rootJobRunKey) {
    this.rootJobRunKey = rootJobRunKey;
    return this;
}
            /**
     * The time at which the cluster validation started in epoch milliseconds.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("clusterValidationStartTime")
private Long clusterValidationStartTime;

        /**
         * The time at which the cluster validation started in epoch milliseconds.
         * @param clusterValidationStartTime the value to set
         * @return this builder
         **/
        

public Builder clusterValidationStartTime(Long clusterValidationStartTime) {
    this.clusterValidationStartTime = clusterValidationStartTime;
    return this;
}
            /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("startTime")
private Long startTime;

        /**
         * The time at which the job execution started in epoch milliseconds.
         * @param startTime the value to set
         * @return this builder
         **/
        

public Builder startTime(Long startTime) {
    this.startTime = startTime;
    return this;
}
            /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("endTime")
private Long endTime;

        /**
         * The time at which the job execution started in epoch milliseconds.
         * @param endTime the value to set
         * @return this builder
         **/
        

public Builder endTime(Long endTime) {
    this.endTime = endTime;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("state")
private State state;



public Builder state(State state) {
    this.state = state;
    return this;
}
            /**
     * The external ID of the task execution.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("externalId")
private String externalId;

        /**
         * The external ID of the task execution.
         * @param externalId the value to set
         * @return this builder
         **/
        

public Builder externalId(String externalId) {
    this.externalId = externalId;
    return this;
}
            /**
     * Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("retryAttempt")
private Integer retryAttempt;

        /**
         * Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.
         * @param retryAttempt the value to set
         * @return this builder
         **/
        

public Builder retryAttempt(Integer retryAttempt) {
    this.retryAttempt = retryAttempt;
    return this;
}


        public TaskRunSummary build() {
            TaskRunSummary model = new TaskRunSummary(this.key
                , this.name
                , this.taskKey
                , this.jobRunKey
                , this.parentJobRunKey
                , this.rootJobRunKey
                , this.clusterValidationStartTime
                , this.startTime
                , this.endTime
                , this.state
                , this.externalId
                , this.retryAttempt);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(TaskRunSummary model) {
                this.key(model.getKey());
    this.name(model.getName());
    this.taskKey(model.getTaskKey());
    this.jobRunKey(model.getJobRunKey());
    this.parentJobRunKey(model.getParentJobRunKey());
    this.rootJobRunKey(model.getRootJobRunKey());
    this.clusterValidationStartTime(model.getClusterValidationStartTime());
    this.startTime(model.getStartTime());
    this.endTime(model.getEndTime());
    this.state(model.getState());
    this.externalId(model.getExternalId());
    this.retryAttempt(model.getRetryAttempt());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * The OCID of the TaskRun.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * The OCID of the TaskRun.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * A user-friendly name. Does not have to be unique, and is changeable.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * The display name of the task. User can specify a value for this.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("taskKey")
    private final String taskKey;

        /**
     * The display name of the task. User can specify a value for this.
     * @return the value
     **/
    
    public String getTaskKey() {
        return taskKey;
    }


        /**
     * The OCID of the job.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("jobRunKey")
    private final String jobRunKey;

        /**
     * The OCID of the job.
     * @return the value
     **/
    
    public String getJobRunKey() {
        return jobRunKey;
    }


        /**
     * The OCID of the job.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("parentJobRunKey")
    private final String parentJobRunKey;

        /**
     * The OCID of the job.
     * @return the value
     **/
    
    public String getParentJobRunKey() {
        return parentJobRunKey;
    }


        /**
     * The OCID of the job.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("rootJobRunKey")
    private final String rootJobRunKey;

        /**
     * The OCID of the job.
     * @return the value
     **/
    
    public String getRootJobRunKey() {
        return rootJobRunKey;
    }


        /**
     * The time at which the cluster validation started in epoch milliseconds.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("clusterValidationStartTime")
    private final Long clusterValidationStartTime;

        /**
     * The time at which the cluster validation started in epoch milliseconds.
     * @return the value
     **/
    
    public Long getClusterValidationStartTime() {
        return clusterValidationStartTime;
    }


        /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("startTime")
    private final Long startTime;

        /**
     * The time at which the job execution started in epoch milliseconds.
     * @return the value
     **/
    
    public Long getStartTime() {
        return startTime;
    }


        /**
     * The time at which the job execution started in epoch milliseconds.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("endTime")
    private final Long endTime;

        /**
     * The time at which the job execution started in epoch milliseconds.
     * @return the value
     **/
    
    public Long getEndTime() {
        return endTime;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("state")
    private final State state;

    
    public State getState() {
        return state;
    }


        /**
     * The external ID of the task execution.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("externalId")
    private final String externalId;

        /**
     * The external ID of the task execution.
     * @return the value
     **/
    
    public String getExternalId() {
        return externalId;
    }


        /**
     * Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("retryAttempt")
    private final Integer retryAttempt;

        /**
     * Sequence number of the current retry attempt. 0 for the original. 1, 2, 3, ... for subsequent retry attempts.
     * @return the value
     **/
    
    public Integer getRetryAttempt() {
        return retryAttempt;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("TaskRunSummary(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", taskKey=").append(String.valueOf(this.taskKey));
        sb.append(", jobRunKey=").append(String.valueOf(this.jobRunKey));
        sb.append(", parentJobRunKey=").append(String.valueOf(this.parentJobRunKey));
        sb.append(", rootJobRunKey=").append(String.valueOf(this.rootJobRunKey));
        sb.append(", clusterValidationStartTime=").append(String.valueOf(this.clusterValidationStartTime));
        sb.append(", startTime=").append(String.valueOf(this.startTime));
        sb.append(", endTime=").append(String.valueOf(this.endTime));
        sb.append(", state=").append(String.valueOf(this.state));
        sb.append(", externalId=").append(String.valueOf(this.externalId));
        sb.append(", retryAttempt=").append(String.valueOf(this.retryAttempt));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof TaskRunSummary)) {
            return false;
        }

        TaskRunSummary other = (TaskRunSummary) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.taskKey, other.taskKey) &&
            java.util.Objects.equals(this.jobRunKey, other.jobRunKey) &&
            java.util.Objects.equals(this.parentJobRunKey, other.parentJobRunKey) &&
            java.util.Objects.equals(this.rootJobRunKey, other.rootJobRunKey) &&
            java.util.Objects.equals(this.clusterValidationStartTime, other.clusterValidationStartTime) &&
            java.util.Objects.equals(this.startTime, other.startTime) &&
            java.util.Objects.equals(this.endTime, other.endTime) &&
            java.util.Objects.equals(this.state, other.state) &&
            java.util.Objects.equals(this.externalId, other.externalId) &&
            java.util.Objects.equals(this.retryAttempt, other.retryAttempt);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.taskKey == null ? 43 : this.taskKey.hashCode());
        result = (result * PRIME) + (this.jobRunKey == null ? 43 : this.jobRunKey.hashCode());
        result = (result * PRIME) + (this.parentJobRunKey == null ? 43 : this.parentJobRunKey.hashCode());
        result = (result * PRIME) + (this.rootJobRunKey == null ? 43 : this.rootJobRunKey.hashCode());
        result = (result * PRIME) + (this.clusterValidationStartTime == null ? 43 : this.clusterValidationStartTime.hashCode());
        result = (result * PRIME) + (this.startTime == null ? 43 : this.startTime.hashCode());
        result = (result * PRIME) + (this.endTime == null ? 43 : this.endTime.hashCode());
        result = (result * PRIME) + (this.state == null ? 43 : this.state.hashCode());
        result = (result * PRIME) + (this.externalId == null ? 43 : this.externalId.hashCode());
        result = (result * PRIME) + (this.retryAttempt == null ? 43 : this.retryAttempt.hashCode());
        return result;
    }


}
